terminus-ui/link/src/link.component.spec.ts

```diff
@@ -1,23 +1,98 @@
+import { APP_BASE_HREF } from '@angular/common';
+import { Component, ViewChild } from '@angular/core';
+import { ComponentFixture, tick } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { RouterModule } from '@angular/router';
+
+import { createComponent } from '@terminus/ngx-tools/testing';
+
 import { TsLinkComponent } from './link.component';
+import { TsLinkModule } from './link.module';
+
+@Component({
+  template: `
+  <ts-link
+    [destination]="destination"
+    [isExternal]="isExternal"
+    [tabIndex]="tabIndex"
+  >
+  My Link Text
+  </ts-link>
+  `,
+})
+class TestHostComponent {
+  destination!: any;
+  isExternal!: boolean;
+  tabIndex!: number | undefined;
+
+  @ViewChild(TsLinkComponent)
+  linkComponent!: TsLinkComponent;
+}
 
 
 describe(`TsLinkComponent`, function() {
-  let component: TsLinkComponent;
+  let component: TestHostComponent;
+  let fixture: ComponentFixture<TestHostComponent>;
+  let link: HTMLElement;
+  let linkComponent: TsLinkComponent;
 
   beforeEach(() => {
-    component = new TsLinkComponent();
+    fixture = createComponent(TestHostComponent, [{provide: APP_BASE_HREF, useValue: '/my/app'}], [TsLinkModule, RouterModule.forRoot([])]);
+    component = fixture.componentInstance;
+    linkComponent = component.linkComponent;
+    fixture.detectChanges();
   });
 
 
-  it(`should exist`, () => {
-    expect(component).toBeTruthy();
+  test(`should exist`, () => {
+    expect(linkComponent).toBeTruthy();
+  });
+
+  describe(`isInternal`, () => {
+    test(`should default and retrieve`, () => {
+      link = fixture.debugElement.query(By.css('.c-link')).nativeElement as HTMLElement;
```

Comment:
Since you're repeating this multiple times, probably good to put in `beforeEach` block.

Reply:
Not a change request, but this does exemplify why I started creating test-helpers for my specs.. I was starting to see these super long queries _all_ over my tests so I wrapped each one up in a function for easy re-use.

Reply:
I had it in the beforeEach but it was not getting updated properly.

Reply:
Might take a look at some existing specs. Some reference: https://github.com/GetTerminus/terminus-ui/blob/master/terminus-ui/button/src/button.component.spec.ts#L52-L58